### PR TITLE
FINERACT-1074: Cleaning up plugin configurations

### PR DESCRIPTION
**Pull request by ptuomola**
## Description
Cleaned up the plugin configuration by ensuring all plugins used are the latest versions, and by pulling each plugin configuration to a separate section in build.gradle, with common parameters (repositories etc) at the start of the file. 

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [x] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [x] API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm has been updated with details of any API changes.

- [x] Integration tests have been created/updated for verifying the changes made.

- [x] All Integrations tests are passing with the new commits.

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

**Review by xurror — APPROVED** [review]

**Comment by xurror** [comment]
Some merge conflicts here. Will merge once they are resolved

**Comment by ptuomola** [comment]
Thanks @xurror - merge conflicts should be all resolved now

File: fineract-provider/build.gradle [hunk 1]
```diff
@@ -20,10 +20,17 @@ description = '''\
 Run as:
 gradle clean bootRun'''
 
+group = 'org.apache.fineract'
+buildDir = new File(rootProject.projectDir, "../build")
+
+repositories {
+    jcenter()
+    mavenCentral()
```
Comment (vorburger) [inline fineract-provider/build.gradle:28]: This shouldn't be required, no? We didn't have it before either

Comment (ptuomola) [reply fineract-provider/build.gradle:28]: Hi @vorburger  - welcome back!

The same lines were there, but just hidden half way down the build.gradle file (lines 164-170 in the previous version). All I did was to reorganise the existing content - I was not trying to make any functional changes at this point, so did not remove any of the settings.

I have another PR ready to further reorganise the content by e.g. grouping all the plugin configurations together. But once we have the contents structured in a more sensible way, then it should be easy to see whether all of the settings are actually needed and remove the ones that are not.

Comment (vorburger) [reply fineract-provider/build.gradle:28]: SGTM. FYI what I meant was specifically the mavenCentral() line - AFAIK jCenter() "proxies" (mirrors) Maven Central already anyway, so this is unnecessary and probably just slows down (failed) look ups.

**Pull request merged**